presto-main/src/main/java/com/facebook/presto/server/ResourceGroupInfo.java

```diff
@@ -0,0 +1,307 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupState;
+import com.facebook.presto.spi.resourceGroups.SchedulingPolicy;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/*
+ * This class is exposed to external systems via ResourceGroupStateInfoResource and QueryStateInfoResource.
```

Comment:
BTW I checked the original diff. This file was originally in package execution, which depends on the package server, not the other way around. So I moved it to spi. It could have been moved to server. But if we have to explicitly put a comment here, there is an argument whether it's better to leave it in SPI (so it's more obvious something external would depend on it). I'll let you decide.

Reply:
Thanks for pointing this out. It is intentional because we are not yet ready to make it an API, but don't want to make changes willy-nilly.